substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/IntrinsificationPluginRegistry.java

```diff
@@ -68,29 +89,41 @@ private static AnalysisMethod toAnalysisMethod(ResolvedJavaMethod method) {
     }
 
     private static final Object NULL_MARKER = new Object();
-
     /**
      * Contains all the elements intrinsified during analysis. Only these elements will be
      * intrinsified during compilation. We cannot intrinsify an element during compilation if it was
      * not intrinsified during analysis since it can lead to compiling code that was not seen during
      * analysis.
      */
-    private final ConcurrentHashMap<CallSiteDescriptor, Object> analysisElements = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<CallSiteDescriptor, Object> globalAnalysisElements = new ConcurrentHashMap<>();
+    final ThreadLocal<ConcurrentHashMap<CallSiteDescriptor, Object>> threadLocalRegistry = new ThreadLocal<>();
 
-    public void add(ResolvedJavaMethod method, int bci, Object element) {
-        Object nonNullElement = element != null ? element : NULL_MARKER;
-        Object previous = analysisElements.put(new CallSiteDescriptor(method, bci), nonNullElement);
+    AutoCloseable startThreadLocalRegistry() {
+        return new AutoCloseable() {
+            {
+                ImageSingletons.lookup(ReflectionPlugins.ReflectionPluginRegistry.class).threadLocalRegistry.set(new ConcurrentHashMap<>());
+            }
+
+            @Override
+            public void close() {
+                ImageSingletons.lookup(ReflectionPlugins.ReflectionPluginRegistry.class).threadLocalRegistry.remove();
+            }
+        };
+    }
```

Comment:
do the same for the intrinsification plugin

Reply:
Doing that could have an impact on your `VarHandle` problem